.github/workflows/test.yml

```diff
@@ -0,0 +1,185 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - ci/test/nightly
+  pull_request:
+
+jobs:
+
+  cache_info:
+    name: Bootstrap cache
+    runs-on: ubuntu-latest
+    outputs:
+      crates-io-index-head: ${{ steps.ls-crates-io-index.outputs.head }}
+      cargo-lock-hash: ${{ steps.hash-cargo-lock.outputs.hash }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - id: ls-crates-io-index
+        name: Get head commit hash of crates.io registry index
+        run: |
+          commit=$(
+            git ls-remote --heads https://github.com/rust-lang/crates.io-index.git master |
+            cut -f 1
+          )
+          echo "$commit"
+          echo "::set-output name=head::$commit"
+
+      - id: hash-cargo-lock
+        name: Calculate dependency cache key
+        run: |
+          hash=$(
+            ci/strip-own-version-from-cargo-lock.pl Cargo.lock |
+            sha1sum | cut -d ' ' -f 1
+          )
+          echo "$hash"
+          echo "::set-output name=hash::$hash"
+
+  update_deps:
+    name: Update dependencies
+    needs: cache_info
+    # Caches on Windows and Unix do not interop:
+    # https://github.com/actions/cache/issues/330#issuecomment-637701649
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    env:
+      RUSTFLAGS: -D warnings
+      RUST_BACKTRACE: 1
+      CARGO_FLAGS: --verbose --locked
+    steps:
+      - name: Cache cargo registry index
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/index
+          key: cargo-index-${{ needs.cache_info.outputs.crates-io-index-head }}
+          restore-keys: cargo-index-
+
+      - id: cargo-deps
+        name: Cache cargo dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry/cache
+            ~/.cargo/git/db
+          key: cargo-deps-v1-${{ needs.cache_info.outputs.cargo-lock-hash }}
+
+      - name: Check out the repository
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Fetch dependencies and update cargo registry
+        run: cargo fetch --locked
+
+  tests:
+    name: Tests
+    needs: [cache_info, update_deps]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        flags: ['', '--release']
+        toolchain: [stable, nightly]
+    steps:
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+          default: true
+          components: rustfmt
```

Comment:
rustfmt shouldn't be needed if you build jormungandr with `--no-default-features`, but then you can only use the `--manifest-path` option, so maybe it's not worth bothering with.